app/src/main/java/org/dhis2/usescases/teiDashboard/TeiDashboardMobileActivity.java

```diff
@@ -471,4 +570,34 @@ public void updateNoteBadge(int numberOfNotes) {
     private int getLastTabPosition() {
         return binding.tabLayout.getTabCount() - 1;
     }
+
+    public LiveData<Boolean> observeGrouping() {
+        return groupByStage;
+    }
+
+    @Override
+    public void setFiltersLayoutState() {
+        filtersShowing.setValue(!filtersShowing.getValue());
+    }
+
+    private void showHideFilters(boolean showFilter) {
+        if (OrientationUtilsKt.isPortrait()) {
+            if (showFilter) {
+                binding.tabLayout.setVisibility(View.GONE);
+                binding.teiPager.setUserInputEnabled(false);
+            } else {
+                binding.tabLayout.setVisibility(View.VISIBLE);
+                binding.teiPager.setUserInputEnabled(true);
+            }
+        }
+    }
```

Comment:
I would rethink this logic in the presenter and move 2 methods to the view, one for showing and another one for hiding. Then you could unit test it

Reply:
Re-done